common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java

```diff
@@ -105,8 +110,11 @@ public void run() {
                 if (privateKey == null) {
```

Comment:
Wouldn't the if statement always be false ? This still needs to be fixed i believe , where we need to set the right thumbprint from the challenge headers and compare with the one in the device certificate

Reply:
"thumbprint from the challenge headers" this only possibly happen when talking to :token endpoint.  This happens when broker receive device certificate challenge request during the token request. 

And currently, the code is missing this part. https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/406 I'm going to fix this in other PR. This PR is focusing on fixing the pkeyauth challenge during the :auth flow.